var/spack/repos/builtin/packages/sollve/package.py

```diff
@@ -0,0 +1,242 @@
+            return
+        codesign = which('codesign')
+        mkdir('tmp')
+        llvm_check_file = join_path('tmp', 'llvm_check')
+        copy('/usr/bin/false', llvm_check_file)
+
+        try:
+            codesign('-f', '-s', 'lldb_codesign', '--dryrun',
+                     llvm_check_file)
+
+        except ProcessError:
+            explanation = ('The "lldb_codesign" identity must be available'
+                           ' to build LLVM with LLDB. See https://llvm.org/'
+                           'svn/llvm-project/lldb/trunk/docs/code-signing'
+                           '.txt for details on how to create this identity.')
+            raise RuntimeError(explanation)
+
+    def setup_build_environment(self, env):
+        env.append_flags('CXXFLAGS', self.compiler.cxx11_flag)
+
+    def setup_run_environment(self, env):
+        if '+clang' in self.spec:
+            env.set('CC', join_path(self.spec.prefix.bin, 'clang'))
+            env.set('CXX', join_path(self.spec.prefix.bin, 'clang++'))
+
+    def cmake_args(self):
+        spec = self.spec
+        cmake_args = [
+            '-DLLVM_REQUIRES_RTTI:BOOL=ON',
+            '-DLLVM_ENABLE_RTTI:BOOL=ON',
+            '-DLLVM_ENABLE_EH:BOOL=ON',
+            '-DCLANG_DEFAULT_OPENMP_RUNTIME:STRING=libomp',
+            '-DPYTHON_EXECUTABLE:PATH={0}'.format(spec['python'].command.path),
+        ]
+
+        # TODO: Instead of unconditionally disabling CUDA, add a "cuda" variant
+        #       (see TODO above), and set the paths if enabled.
+        cmake_args.extend([
+            '-DCUDA_TOOLKIT_ROOT_DIR:PATH=IGNORE',
+            '-DCUDA_SDK_ROOT_DIR:PATH=IGNORE',
+            '-DCUDA_NVCC_EXECUTABLE:FILEPATH=IGNORE',
+            '-DLIBOMPTARGET_DEP_CUDA_DRIVER_LIBRARIES:STRING=IGNORE'])
+
+        if '+gold' in spec:
+            cmake_args.append('-DLLVM_BINUTILS_INCDIR=' +
+                              spec['binutils'].prefix.include)
+        if '+polly' in spec:
+            cmake_args.append('-DLINK_POLLY_INTO_TOOLS:Bool=ON')
+        else:
+            cmake_args.extend(['-DLLVM_EXTERNAL_POLLY_BUILD:Bool=OFF',
+                               '-DLLVM_TOOL_POLLY_BUILD:Bool=OFF',
+                               '-DLLVM_POLLY_BUILD:Bool=OFF',
+                               '-DLLVM_POLLY_LINK_INTO_TOOLS:Bool=OFF'])
+
+        if '+python' in spec and '+lldb' in spec:
+            cmake_args.append('-DLLDB_USE_SYSTEM_SIX:Bool=TRUE')
+        if '+clang' not in spec:
+            cmake_args.append('-DLLVM_EXTERNAL_CLANG_BUILD:Bool=OFF')
+        if '+lldb' not in spec:
+            cmake_args.extend(['-DLLVM_EXTERNAL_LLDB_BUILD:Bool=OFF',
+                               '-DLLVM_TOOL_LLDB_BUILD:Bool=OFF'])
+        if '+lld' not in spec:
+            cmake_args.append('-DLLVM_TOOL_LLD_BUILD:Bool=OFF')
+        if '+internal_unwind' not in spec:
+            cmake_args.append('-DLLVM_EXTERNAL_LIBUNWIND_BUILD:Bool=OFF')
```

Comment:
Should all of these things be explicitly enabled in an else clause?

Reply:
Agree. This shouldn't rely on the "default" CMake options. I add else clauses and explicitly specify CMake options in them.